packages/forms/src/patterns/select-dropdown/select-dropdown.test.ts

```diff
@@ -0,0 +1,161 @@
+import { describe, expect, it } from 'vitest';
+import {
+  createSchema,
+  selectDropdownConfig,
+  type SelectDropdownPattern,
+} from './select-dropdown';
+
+describe('SelectDropdownPattern tests', () => {
+  describe('createSchema', () => {
+    it('should create schema for required dropdown', () => {
+      const data: SelectDropdownPattern['data'] = {
+        label: 'Test Label',
+        required: true,
+        options: [
+          { value: 'value1', label: 'Option 1' },
+          { value: 'value2', label: 'Option 2' },
+        ],
+      };
+
+      const schema = createSchema(data);
+      expect(schema.safeParse('value1').success).toBe(true);
+      expect(schema.safeParse('value2').success).toBe(true);
+      expect(schema.safeParse('invalid').success).toBe(false);
+      expect(schema.safeParse('').success).toBe(false);
+      expect(() => schema.parse('')).toThrow();
+    });
+
+    it('should create schema for optional dropdown', () => {
+      const data: SelectDropdownPattern['data'] = {
+        label: 'Test Label',
+        required: false,
+        options: [
+          { value: 'value1', label: 'Option 1' },
+          { value: 'value2', label: 'Option 2' },
+        ],
+      };
+
+      const schema = createSchema(data);
+      expect(schema.safeParse('value1').success).toBe(true);
+      expect(schema.safeParse('value2').success).toBe(true);
+      expect(schema.safeParse('invalid').success).toBe(false);
+      expect(schema.safeParse('').success).toBe(true);
+    });
+
+    it('should throw error if no options are provided', () => {
+      const data: SelectDropdownPattern['data'] = {
+        label: 'Test Label',
+        required: true,
+        options: [],
+      };
+
+      expect(() => createSchema(data)).toThrow(
+        'Options must have at least one value'
+      );
+    });
+  });
+
+  describe('selectDropdownConfig', () => {
+    it('should parse user input correctly', () => {
+      const pattern: SelectDropdownPattern = {
+        type: 'selectDropdown',
+        id: 'test',
+        data: {
+          label: 'Test Dropdown',
+          required: true,
+          options: [
+            { value: 'value1', label: 'Option 1' },
+            { value: 'value2', label: 'Option 2' },
+          ],
+        },
+      };
+
+      const inputValue = 'value1';
+      if (selectDropdownConfig.parseUserInput) {
```

Comment:
You could also do this here, to avoid skipping the test, in the event parseUserInput becomes undefined:

```typescript
if (!selectDropdownConfig.parseUserInput) {
    expect.fail('selectDropdownConfig.parseUserInput is not undefined');
}
```

Then the test code, below, will be type-safe without the if condition.

Reply:
@danielnaab do you mean something like this?

```
      if (!selectDropdownConfig.parseUserInput) {
        expect.fail('selectDropdownConfig.parseUserInput is not undefined');
      }
      const result = selectDropdownConfig.parseUserInput(pattern, inputValue);
      console.log('Test parse result:', result);
      if (result.success) {
        expect(result.data).toBe('value1');
      } else {
        throw new Error('Unexpected validation failure');
      }
```